integrations/cross_browsers/BrowserStack_Library.py

```diff
@@ -60,11 +61,61 @@ def get_active_session_details(self):
 
         return session_details
 
-
-    def get_session_logs(self, timeout=10):
-        "Return the session log in text format"
-        session_details = self.get_active_session_details()
-        session_log_url = session_details['logs']
-        session_log = requests.get(f'{session_log_url}',auth=self.auth,timeout=timeout).text
-
-        return session_log
+    def extract_session_id(self, session_url):
+        "Extract session id from session url"
+        import re
+        # Use regex to match the session ID, which is a 40-character hexadecimal string
+        match = re.search(r'/sessions/([a-f0-9]{40})', session_url)
+        if match:
+            return match.group(1)
+        else:
+            return None
+
+    def upload_terminal_logs(self, file_path, session_id = None, appium_test = False, timeout=30):
+        "Upload the terminal log to BrowserStack"
+        try:
+            # Get session ID if not provided
+            if session_id is None:
+                session_details = self.get_active_session_details()
+                session_id = session_details['hashed_id']
+                if not session_id:
+                    raise ValueError("Session ID could not be retrieved. Check active session details.")
+
+            # Determine the URL based on the type of test
+            if appium_test:
+                url = f'{self.browserstack_cloud_api_server_url}/app-automate/sessions/{session_id}/terminallogs'
+            else:
+                url = f'{self.browserstack_cloud_api_server_url}/automate/sessions/{session_id}/terminallogs'
+
+            # Open the file using a context manager to ensure it is properly closed
+            with open(file_path, 'rb') as file:
+                files = {'file': file}
+                # Make the POST request to upload the file
+                response = requests.post(url, auth=self.auth, files=files, timeout=timeout)
+
+            # Check if the request was successful
+            if response.status_code == 200:
```

Comment:
we can remove this check and add `response.raise_for_status()`, this way, it would automatically be handled in the exceptions if status fails

Reply:
@akkuldn In our scenario, response.raise_for_status() is not useful as it immediately raise an exception and exit that block of code. We need to return the response for logging. And one more thing response.raise_for_status() consider 2xx and 3xx responses as success which is not true in our scenario and we need more control over specific HTTP responses.